service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageQueries.kt

```diff
@@ -363,6 +367,49 @@ fun Database.Read.getReceiversForNewMessage(
         }
 }
 
+fun Database.Read.isUserAuthorizedToSendTo(user: AuthenticatedUser, accountIds: Set<UUID>): Boolean {
+    // language=SQL
+    val sql = """
+        WITH children AS (
+            SELECT child_id
+            FROM child_acl_view
+            WHERE employee_id = :employeeId
+        ), receivers AS (
+            SELECT g.guardian_id AS receiver_id
+            FROM children c
+            JOIN guardian g ON g.child_id = c.child_id
+            WHERE NOT EXISTS (
+                SELECT 1 FROM messaging_blocklist bl
+                WHERE bl.child_id = c.child_id
+                AND bl.blocked_recipient = g.guardian_id
+            )
+
+            UNION DISTINCT
+
+            SELECT fc.head_of_child AS receiver_id
+            FROM children c
+            JOIN fridge_child fc ON fc.child_id = c.child_id AND daterange(fc.start_date, fc.end_date, '[]') @> :date
+            WHERE NOT EXISTS (
+                SELECT 1 FROM messaging_blocklist bl
+                WHERE bl.child_id = c.child_id
+                AND bl.blocked_recipient = fc.head_of_child
+            )
+        ), authorized_accounts AS (
+            SELECT acc.id as account_id
+            FROM receivers r
+            JOIN message_account acc ON acc.person_id = r.receiver_id
+        )
+        SELECT :accountIds <@ ARRAY(SELECT account_id FROM authorized_accounts)
```

Comment:
I thought that it's better to keep this check inside Postgres instead of fetching potentially hundreds of account ids to Kotlin side and doing the set comparison there.

Reply:
Are there any performance implications with this approach? I do not know how pg optimizes these kind of queries, but wouldn't it be potentially performance wise better to just inner join all accounts that match the criteria and check that the array length matches?

Reply:
Now checking the length instead.